packages/vote/vote_context.go

```diff
@@ -74,3 +78,29 @@ func (vc *Context) IsNew() bool {
 func (vc *Context) HadFirstRound() bool {
 	return vc.Rounds == 1
 }
+
+// HadFixedRound tells whether the vote context is in the last l2 rounds of fixed threshold
+func (vc *Context) HadFixedRound(coolingOffPeriod int, finalizationThreshold int, fixedEndingThreshold int) bool {
+	consecutiveRounds := finalizationThreshold - fixedEndingThreshold // l - l2
+	// check whether we have enough opinions to say whether this vote context is finalized.
+	if len(vc.Opinions[1:]) < coolingOffPeriod + consecutiveRounds {
+		return false
+	}
+	// check weather we have enough opinions and parameters are valid
+	if len(vc.Opinions) < consecutiveRounds || consecutiveRounds < 0 {
```

Comment:
what is the reason that above we check vc.Opinions[1:] but here vc.Opinions ?

Reply:
why the case consecutiveRounds < 0 ? Since this is a parameter check should we check it when we initialise the parameters?

Reply:
About parameters wasn't sure where to put the check as no other parameters have been checked.  They are provided in DefaultParameters() but what when we put wrong values in tests?
For vc.Opinions[1:] I took it from isFinalized(). but it can be misleading. Maybe I'll simplify it also there to be consistent

Reply:
so from our discussion i think it makes sense to ignore failures in parameters input. rather we add a comment to the parameter default description, where we say which range makes sense. This should be in a different PR though, once we merged our current PRs